### PR TITLE
small script for testing all given SMTP and IMAP servers

### PR DESCRIPTION
**Pull request by missytake**
A testing script for automatically testing whether there is an actual SMTP/IMAP server listening on the given ports :) See #219 for context.

Also introduces the `skip_auto_test` provider setting, which exempts them from CI testing through this script.

### CODE REVIEW

File: testscript.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                    test_smtp(server)
+                if server["type"] == "imap":
+                    test_imap(server)
+            except Exception as e:
+                print("[error] %s:%s \t%s: %s" %
+                      (server["hostname"], server["port"], sys.exc_info()[0].__name__, sys.exc_info()[1]))
+                exitcode += 1
+
+    if __name__ == "__main__":
+        exit(exitcode)
```
Comment (missytake) [inline testscript.py:104]: Not sure if this is too bold :D but I mean the worst thing which can happen is that it crashes at exit ;)

**Comment by r10s** [comment]
great! that's very useful! can we execute that regularly from ci? just on any or would probably be frequently enough so that we catch errors - also that would test the changes in the pr then.

**Review by r10s — APPROVED** [review]
works to move forward, we can do additional changes and merge that just in, it is already useful as is when staring manually.

**Comment by missytake** [comment]
So there is a github action now :) I also tracked down all the errors the script was throwing. Now we can use this for CI :)

One interesting thing would be to let the tests run in parallel so they don't take 2 minutes. But I guess that would just be some optimization :shrug:

**Comment by r10s** [comment]
yeah!

**Pull request merged**